Feature/shell read

jted0537 opened the pull request:
- shell read함수 기능구현
- read 관련 valid / invalid input test code 추가
- read and write test code 추가
- 테스트용 변수들 상수로 추가

jted0537 committed:
feature: add readtest invalid input

jted0537 committed:
feature: add read mock test

jted0537 committed:
[refactoring] change return type to string

jted0537 committed:
[refactoring] remove main code

jted0537 committed:
[feature] add read_and_write test

jted0537 committed:
[refactoring] remove result.txt

jted0537 committed:
refactoring: introduce constant

parkvibes commented on test_shell_app/shell.py:
```diff
@@ -1,9 +1,29 @@
+            _, stderr = Popen(
+                f"ssd R {addr}", shell=True, stdout=PIPE, stderr=PIPE
+            ).communicate()
+            if stderr != "":
+                raise Exception("stderr")
```
Exception을 이용해 `stderr` 내용을 출력해주면 무슨 error가 발생했는지 더욱 명확해 질 것 같습니다.

jted0537 replied:
넵 Detail 한 error 내역도 출력될 수 있게 다음 PR에 반영하겠습니다.
좋은 의견 감사합니다!

parkvibes commented on test_shell_app/shell.py:
```diff
@@ -1,9 +1,29 @@
 
     def read(self, addr):
-        pass
+        if addr < 0 or addr > 99:
+            print("INVALID PARAMETER", flush=True)
+            return ""
```
이건 `write()` 함수에서도 사용할 수 있도록 함수로 빼는 건 어떨까요?

jted0537 replied:
넵 현재 해당 내용을 고려중인데, 현재 파라미터가
```
- Write의 경우 1st: addr, 2nd: val
- Read의 경우 1st: addr
- FullWrite의 경우 1st: val 
```
인 점을 고려해서 해당함수 제작하신 분들과 상의해봐야할 것 같습니다. 아이디어 감사드립니다.

jted0537 committed:
[refactoring] add constant for temp variable

parkvibes reviewed (commented):
test case가 빈틈이 없어보입니다.
코멘트 한번 확인 부탁드려요~

jted0537 committed:
B
Merge branch 'main' of https://github.com/jted0537/CRA_SSDProject into feature/shell_read

jted0537 committed:
[refactoring] remove useless variable

yezinMong reviewed (approved):
변수를 깔끔하게 잘 정리하신 것 같습니다!
write 테스트 케이스 작성할 때 참고하도록 하겠습니다.😊

sw2kang reviewed (approved):
shell read 부분 깔끔하게 작성 하셨네요~ 점점 좋은 코드가 되는 것 같습니다.
